spec/requests/manager_edit_superintendent_spec.rb

```diff
@@ -0,0 +1,110 @@
+require 'rails_helper'
+
+describe 'Manager edits superintendent' do
+  context 'GET /condos/:condo_id/superintendents/new' do
+    it 'must be authenticated to edit an superintendent' do
+      condo = create :condo, name: 'Condomínio X'
+      tower = create(:tower, condo:)
+      unit11 = tower.floors.first.units.first
```

Comment:
Depois do PR do Rulian, já é possível usar a factory create(:resident, :with_residence, condo:) que já abstrai a criação de torre quando é desnecessária, e é possível pegar a residência do morador normalmente com resident.residence.

Me parece ser o caso aqui, já que o resident está sendo criado para que possa ser promovido a superintendent.

Não vou comentar nos demais, mas a lógica se aplica onde mais for cabível.